lms/djangoapps/instructor/views/data_access.py

```diff
@@ -0,0 +1,374 @@
+    """
+    ids_in_course = CourseEnrollment.objects.filter(course_id=course_id,
+                                                    is_active=1,
+                                                    ).values_list(DatabaseFields.USER_ID)
+    total_students = User.objects.filter(id__in=ids_in_course)
+    without_open = total_students.exclude(id__in=
+                                          StudentModule.objects.filter(
+                                              module_state_key=query.entity_id,
+                                              course_id=course_id).values_list(DatabaseFields.STUDENT_ID),
+                                          )
+    return process_results(course_id, without_open, DatabaseFields.ID, DatabaseFields.EMAIL)
+
+
+def not_completed_query(course_id, query):
+    """
+    Specific db query for sections or problems that are not completed
+    """
+    ids_in_course = CourseEnrollment.objects.filter(course_id=course_id,
+                                                    is_active=1,
+                                                    ).values_list(DatabaseFields.USER_ID)
+    total_students = User.objects.filter(id__in=ids_in_course)
+    without_completed = total_students.exclude(id__in=
+                                               StudentModule.objects.filter(
+                                                   module_state_key=query.entity_id,
+                                                   course_id=course_id).filter(~Q(grade=None))
+                                               .values_list(DatabaseFields.STUDENT_ID),
+                                               )
+    return process_results(course_id, without_completed, DatabaseFields.ID, DatabaseFields.EMAIL)
+
+
+def completed_query(course_id, query):
+    """
+    Specific db query for sections or problems that are completed
+    """
+    queryset = StudentModule.objects.filter(module_state_key=query.entity_id,
+                                            course_id=course_id,
+                                            ).filter(~Q(grade=None))
+    return process_results(course_id, queryset, DatabaseFields.STUDENT_ID, DatabaseFields.STUDENT_EMAIL)
+
+
+def open_query(course_id, query):
+    """
+    Specific db query for sections or problems that are opened
+    """
+    queryset = StudentModule.objects.filter(module_state_key=query.entity_id, course_id=course_id)
+    return process_results(course_id, queryset, DatabaseFields.STUDENT_ID, DatabaseFields.STUDENT_EMAIL)
+
+
+def process_results(course_id, queryset, id_field, email_field):
+    """
+    Handles any intermediate filtering between specific query criteria and returning to the user
+    """
+    filtered_query = filter_out_students_negative(course_id, queryset)
+    values = filtered_query.values_list(id_field, email_field).distinct()
+    return values
+
+
+def filter_out_students_negative(course_id, queryset):
```

Comment:
write brief comment about why pos/neg cases need to be separate (the db table originating the QuerySet is different)